### ISSUE TITLE
Upload progress with Redux Toolkit

### ISSUE THREAD

**Issue by Steinarthor** [reduxjs/redux-toolkit#2327]
Hi there!

I'm trying to create a progress bar while uploading some data. I can't figure out how to get the current state of the upload progress

I did try to read from the body, but the Readable stream is always locked. 

**Example: PUT request uploading some data**
```
 addCommentAttachmentData: builder.mutation<Record<string, any> | null, { attachmentId: number | undefined; file: FormData } ({
    query: ({ attachmentId, file }) => {
        return {
            url: `/comment-attachments/${attachmentId}/data`,
            method: 'PUT',
            body: file,
            headers: {
                'Content-Type': undefined
            }
        };
    },
   transformResponse: (response, meta) => {
        meta.response.body.getReader().read()....  <-- Always locked.
    },
    invalidatesTags: ['attachments']
})
```
Any help would be greatly greatly appreciated on how one can create an upload progress bar with Redux Toolkit. 🙏

**Comment by markerikson** [comment]
Hmm. I'm not sure this is really feasible atm.

The biggest thing is that RTKQ uses `fetch` by default, via `fetchBaseQuery`.  I just did some checking to confirm, and I don't think that `fetch` supports any form of upload progress yet.

Closest examples I found were https://ilikekillnerds.com/2020/09/file-upload-progress-with-the-fetch-api-is-coming/ and https://javascript.info/fetch-progress , which both suggest that while there's some proof-of-concept prototypes in Chrome, there's nothing formalized and official yet.
